### PR TITLE
Adopt Psalm

### PR DESCRIPTION
**Pull request by Fantom409**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️/❌
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

File: src/Mutex.php [hunk 1]
```diff
@@ -40,6 +42,9 @@ public function __construct(bool $autoRelease = true)
         if ($autoRelease) {
             $locks = &$this->locks;
             register_shutdown_function(function () use (&$locks) {
+                /**
+                 * @var string $lock
```
Comment (samdark) [inline src/Mutex.php:46]: Instead of typing here, can you do it in property declaration?

```php
@psalm-var list<string>
```

File: src/Mutex.php [hunk 2]
```diff
@@ -40,6 +42,9 @@ public function __construct(bool $autoRelease = true)
         if ($autoRelease) {
             $locks = &$this->locks;
             register_shutdown_function(function () use (&$locks) {
```
Comment (samdark) [inline src/Mutex.php:44]:
```suggestion
            register_shutdown_function(function () use (array &$locks) {
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Mutex.php (hunk 1 above)
Comment (Fantom409) [reply src/Mutex.php:46]: Docblock for locks should be enough - it works well on psalm 3.15, but psalm 4 generates an error when variable passed by reference

Example without reference
https://psalm.dev/r/01bd094997

File: src/Mutex.php (hunk 2 above)
Comment (Fantom409) [reply src/Mutex.php:44]: It will cause ParseError

**Pull request merged**